Library/Formula/llvm.rb

```diff
@@ -116,18 +149,49 @@ def install
     # Apple's libstdc++ is too old to build LLVM
     ENV.libcxx if ENV.compiler == :clang
 
-    if build.with?("lldb") && build.without?("clang")
-      raise "Building LLDB needs Clang support library."
-    end
+    (buildpath/"tools/clang").install resource("clang") if build.with? "clang"
 
-    if build.with? "clang"
-      (buildpath/"projects/libcxx").install resource("libcxx")
-      (buildpath/"tools/clang").install resource("clang")
-      (buildpath/"tools/clang/tools/extra").install resource("clang-tools-extra")
+    if build.with? "clang-extra-tools"
+      odie "clang's extra tools must be built with clang" if build.without? "clang"
```

Comment:
I would prefer if you could refer to the various components as `Clang`, `LLDB`, etc. like in the option descriptions. (This also applies to all other `odie` statements.) Alternatively, you could write something like `--with-clang-extra-tools requires --with-clang`, which is somewhat more technical, but has the advantage of being very clear (and a user can copy & paste the option to retry the installation).